src/loadsave.cc

```diff
@@ -2548,23 +2546,6 @@ static bool_ do_savefile_aux(ls_flag_t flag)
 		do_fate(i, flag);
 	}
 
-	/* Load the Traps */
-	if (flag == ls_flag_t::SAVE) tmp16u = max_t_idx;
```

Comment:
I'll see if I can make that work. Compatibility with legacy save files is not something I've paid much attention to unfortunately.